NACT/attention_query_utilities/attention_query.py

```diff
@@ -441,85 +441,89 @@ def set_scanpy_object(self, new_data):
         self.data = new_data
 
 
-# TODO: @Oscar: could you please take care of the clean up and docstrin of this
-# method please?
+# TODO: @Ali: could you take a look at this and see if there are any changes needed. 
 
-    def gene_tfidf(self, top_n_genestrings_dict: dict = None):
-        """Calculate TF-IDF values for all genes in a top-n ranked dictionary.
+    def calculate_tfidf(top_n_names, top_n_df_dict, n_genes=25):
+        """
+        This function receives a pandas series that contains a list of genes in each row, and returns a pandas series with the Corpus-Wide tf-idf scores for each gene.
+        Where the Corpus-Wide tf-idf score for each gene is defined as (1 - Sum(tf-idf)/log(N)), where N is the total number of pseudo documents.
+        The returned series is indexed by the genes and sorted in descending order by the tf-idf scores.
 
-        ***
-        @Oscar: This needs to be cleaned up a bit :)
-        ***
+        
+        Args:
+            top_n_names: A dictionary mapping containing the name of top n genes for each
+            cluster.
+            top_n_df_dict: A dictionary containing the sum of gene attention scores for each
+            cluster.
+            n_genes: A number indicating the length for return values.
 
+        Returns:
+            dataframe: A pandas dataframe with the Corpus-Wide tf-idf scores for each gene, indexed by the genes, sorted in descending order by the tf-idf scores.
+
+        Raises:
+        None
```

Comment:
**R:** Indentation and a period at the end (see [SG](https://google.github.io/styleguide/pyguide.html#32-line-length))